crates/vm/levm/src/opcodes.rs

```diff
@@ -358,12 +358,12 @@ impl From<Opcode> for usize {
 
 /// Represents an opcode function handler.
 #[derive(Debug, Clone, Copy)]
-pub(crate) struct OpCodeFn<'a>(fn(&'_ mut VM<'a>) -> Result<OpcodeResult, VMError>);
+pub(crate) struct OpCodeFn<'a>(fn(&'_ mut VM<'a>) -> Result<bool, VMError>);
```

Comment:
Let's explain in the documentation comment that the boolean returned by the result is `true` if a halt was produced and the caller needs to handle it.